libexec/python/docker/api.py

```diff
@@ -47,24 +52,46 @@ def create_runscript(cmd,base_dir):
     return output_file
 
 
-def get_token(repo_name,namespace="library",scope="repository",permission="pull"):
+def get_token(registry,repo_name,namespace,scope="repository",permission="pull",auth=None):
     '''get_token will use version 2.0 of Docker's service to return a token with given permission and scope - this
```

Comment:
Yeah, I believe `get_token` is more an utility function. It should only be called from `get_config`, `get_manifest, `get_layer`, etc. And these functions can indeed default to `index.docker.io`, but no need to do it twice IMO.
But yeah, it's easy to fix :)